build-tools/packages/version-tools/src/versions.ts

```diff
@@ -189,3 +189,23 @@ function getVersionsFromStrings(prefix: TagPrefix, tags: string[]): string[] {
 	semver.sort(versions);
 	return versions;
 }
+
+/**
+ * Checks if the provided version is a test version.
+ *
+ * Test versions are generated from test/ branches and are published to the test feed.
+ *
+ * @param version - The version to check
+ * @returns - True if the version string is a test version, otherwise false
+ *
+ * @example
+ * returns true
+ * isTestVersion("0.0.0-260312-test");
+ *
+ * @example
+ * returns false
+ * isTestVersion("2.1.0-260312");
+ */
+export function isTestVersion(version: string): boolean {
+	return version.includes("test");
```

Comment:
This is not enough. We cannot just do string testing. If we are creating a new version scheme we need to add test cases, parse the string into its parts, evaluate if the whole string matches the format, etc. I'll try to look at this and the issues @tyler-cai-microsoft has been seeing early next week.

One place to start is finding the code that generates test versions today and refactoring it to not only create test versions but also parse them.